src/bam-utils.js

```diff
@@ -20,46 +24,74 @@ Object.keys(PILEUP_COLORS).map((x, i) => {
   return null;
 });
 
+export const cigarTypeToText = (type) => {
+  if(type === "D"){
+    return "Deletion";
+  } else if(type === "S"){
+    return "Soft clipping";
+  } else if(type === "H"){
+    return "Hard clipping";
+  } else if(type === "I"){
+    return "Insertion";
+  } else if(type === "N"){
+    return "Skipped region";
+  } else {
+    return type;
+  }
+}
+
 export const parseMD = (mdString, useCounts) => {
-  let currPos = 1;
+  let currPos = 0;
   let currNum = 0;
+  let deletionEncountered = false;
+  let bamSeqShift = 0;
   const substitutions = [];
 
   for (let i = 0; i < mdString.length; i++) {
     if (mdString[i].match(/[0-9]/g)) {
       // a number, keep on going
       currNum = currNum * 10 + +mdString[i];
+      deletionEncountered = false;
+    }else if(mdString[i] === "^"){
+      deletionEncountered = true;
     } else {
-      currPos += currNum;
 
+      currPos += currNum;
+      
       if (useCounts) {
         substitutions.push({
           length: currNum,
           type: mdString[i],
         });
+      } else if(deletionEncountered){
+        // Do nothing if there is a deletion and keep on going. Note that there can be multiple deletions "^ATC"
+        // Deletions are visualized using the CIGAR string
+        // However, keep track of where in the bam seq we need to pull the variant.
+        bamSeqShift -= 1;
       } else {
         substitutions.push({
           pos: currPos,
-          base: mdString[i + 0],
+          base: mdString[i],
           length: 1,
+          bamSeqShift: bamSeqShift,
         });
       }
 
       currNum = 0;
       currPos += 1;
     }
   }
-
+  // if(!useCounts){
+  //   console.log(mdString, substitutions)
+  // }
+  
   return substitutions;
 };
 
-export const getSubstitutions = (segment) => {
+export const getSubstitutions = (segment, seq) => {
```

Comment:
What is seq? I've set a bad precedent here but docstrings would be good to have.

Reply:
Done